codegen/src/utils/fetch_metadata.rs

```diff
@@ -97,29 +94,29 @@ pub async fn fetch_metadata_hex(
 }
 
 async fn fetch_metadata_ws(
-    url: &Uri,
+    url: Url,
     version: MetadataVersion,
 ) -> Result<Vec<u8>, FetchMetadataError> {
     let (sender, receiver) = WsTransportClientBuilder::default()
-        .build(url.to_string().parse::<Uri>().unwrap())
```

Comment:
I removed this, it doesn't make sense to parse to URL twice instead better to just clone the URL?! :)

Thus, all APIs are changed to accept to URL by value instead of reference...

Reply:
Hehe yeah, no idea why we were doing that :D